oodeel/datasets/data_handler.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+# Copyright IRT Antoine de Saint Exupéry et Université Paul Sabatier Toulouse III - All
+# rights reserved. DEEL is a research program operated by IVADO, IRT Saint Exupéry,
+# CRIAQ and ANITI - https://www.deel.ai/
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from abc import ABC
+from abc import abstractmethod
+
+import numpy as np
+
+from ..types import Any
+from ..types import Callable
+from ..types import Optional
+from ..types import Tuple
+
+
+class DataHandler(ABC):
+    @abstractmethod
+    @staticmethod
+    def load_dataset(dataset_id: Any, load_kwargs: dict = {}):
+        """Load dataset from different manners
+
+        Args:
+            dataset_id (Any): dataset identification
+            load_kwargs (dict, optional): Additional loading kwargs. Defaults to {}.
+
+        Returns:
+            Any: dataset
+        """
+        raise NotImplementedError()
```

Comment:
This method is designed to use othe methods from the same object so it should not be static. See comment on file torch_data_handler.py on line 176.

Reply:
+1